File: nigeria_compliance_client/nigeria_compliance_client/hooks/supplier.py
import frappe

def validate(doc, method=None):
    if doc.custom_b2b_supplier and doc.custom_b2c_supplier:
        frappe.throw("B2B Supplier and B2C Supplier cannot be same")

    if doc.custom_b2b_supplier:
        check_address_for_b2b_supplier(doc)

def check_address_for_b2b_supplier(doc):
    if not frappe.db.exists("Dynamic Link", {"link_doctype": "Supplier", "link_name": doc.name, "parenttype": "Address" }):
        frappe.throw("Address is required for B2B Supplier")

File: nigeria_compliance_client/public/js/supplier.js
frappe.ui.form.on('Supplier', {
    refresh(frm) {
        // your code here
    },
    custom_b2b_supplier(frm) {
        if (frm.doc.custom_b2b_supplier == 1) {
            if (frm.doc.custom_b2b_supplier == 1 && frm.doc.custom_b2c_supplier == 1) {
                frm.set_value("custom_b2c_supplier", 0)
            }
        }

    },
    custom_b2c_supplier(frm) {
        if (frm.doc.custom_b2c_supplier == 1) {
            if (frm.doc.custom_b2c_supplier == 1 && frm.doc.custom_b2b_supplier == 1) {
                frm.set_value("custom_b2b_supplier", 0)
            }
        }
    }
})